core/admin/port_management.py

```
@@ -0,0 +1,449 @@
+        }
+    
+    # Validate port still available
+    if not validate_port_available(external_port):
+        raise ValueError(f"Port {external_port} is not available")
+    
+    # Get server IP
+    server_ip = get_server_ip()
+    
+    # Add iptables rule to accept traffic on external port
+    try:
+        subprocess.run([
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)